File: Concurrente/Threads1/Java/Threads1.java
package Threads1.Java;

public class Threads1 implements Runnable {
    static final int THREADS = 4;
    int id;

    public Threads1(int id) {
        this.id = id;
    }

    @Override
    public void run() {
        System.out.printf("Hi, I'm thread %d\n", id);
    }

    public static void main(String[] args) throws InterruptedException {
        Thread[] threads = new Thread[THREADS];
        int i;
        
        for (i=0; i< THREADS; i++) {
            threads[i] = new Thread(new Threads1(i));
            threads[i].start();
        }

        for (i=0; i< THREADS; i++) {
            threads[i].join();
        }
        
    }
}

File: Concurrente/Threads1/Python/Threads1.py
import threading, time

THREADS = 4

def thread():
    time.sleep(0.5)
    print("Hi, I'm the thread {}".format(threading.current_thread().name))

def main():
    threads = []

    for i in range(THREADS):
        # Create new threads
        t = threading.Thread(target=thread)
        threads.append(t)
        t.start() # start the thread

    # Wait for all threads to complete
    for t in threads:
        t.join()

    print("End")

if __name__ == "__main__":
    main()